### PR TITLE
Update README

### LINKED ISSUE #248: [DOC] Readme Suggestions
I was trying to build `ucx` using dask and ran into these issues: 

### 1. **Incorrect Dask URL:**
The below url for dask is incorrect i think as it should point to dask instead of rapidsai:

https://github.com/rapidsai/ucx-py/blob/branch-0.10/README.md#dask
```
    git clone git@github.com:rapidsai/dask.git
```

### 2. **http instead of git@**
Also, i am wondering if it will be better to have a `http` urls instead of `git@url` in docs as users wont always have `ssh` keys setup across containers. 

### 3. Dask-cuda conda seems to be pulling previous non ucx-version
Below seems to be pulling previous version of dask-cuda, we may want to add the installation command with with all the dependencies.

Instead of below:
```
conda install -c rapidsai dask-cuda
```

May be something like:
```
conda install -c rapidsai-nightly/label/cuda10.0 -c rapidsai -c nvidia -c numba -c conda-forge dask-cuda
```

### 4. Script to check if ucx is working for a non source installation

This issue may be similar to document UCX usage with dask issue but it will be nice to have test/script i can run which checks if ucx was set up correctly. 

https://github.com/rapidsai/ucx-py/issues/229

### PR DESCRIPTION
**Pull request by quasiben**
PR resolves #248 

@jakirkham  do you have any comments while I'm mucking around in the README ?

cc @VibhuJawa

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,21 +34,21 @@ These three libraries provide a powerful combination of HPC message passing tool
...
-    git clone git@github.com:dask/distributed.git
+    git clone https://github.com:dask/distributed.git
     cd distributed
     pip install -e .
     cd ..
 
 ### dask-cuda ###
 
-    conda install -c rapidsai dask-cuda
+    conda install -c rapidsai-nightly dask-cuda
```
Comment (VibhuJawa) [inline README.md]: I think we would have to add other channels here too as this still seems to pulling a version from `19-09-24`
i.e it pulls below version:
```rapidsai-nightly/label/cuda10.0/linux-64::dask-cuda-0.10.0a190924-py37_15```

While below pulls the latest version:

```bash
conda install -c rapidsai-nightly/label/cuda10.0 -c rapidsai -c nvidia -c numba -c conda-forge dask-cuda
```

Comment (jakirkham) [reply README.md]: I don't think we want to use the CUDA labeled channels. Those are getting phased out. Also it locks in a specific CUDA Version, which the user may not want.

My guess is if this was needed there is another issue occurring here like packages are missing or something similar. Could you please raise a separate issue about this? Possibly on dask-cuda?

Comment (VibhuJawa) [reply README.md]: Yup, 
I forgot to remove the `cuda` version. 
Below works too. 

```
 conda install -c rapidsai-nightly -c rapidsai -c nvidia -c numba -c conda-forge dask-cuda --dry
```
Sure, I will raise the issue on  `dask-cuda` .

**Comment by jakirkham** [comment]
Would add the [`compilers`]( https://github.com/conda-forge/compilers-feedstock ) package to the conda-forge dependencies for building from source.

**Comment by jakirkham** [comment]
Also there is another git URL in the ucx-py section that we might want to fix.

File: README.md [hunk 2]
```diff
@@ -58,15 +58,11 @@ These three libraries provide a powerful combination of HPC message passing tool
 
     git clone https://github.com/openucx/ucx
     cd ucx
-    git remote add Akshay-Venkatesh git@github.com:Akshay-Venkatesh/ucx.git
-    git remote update Akshay-Venkatesh
-    git checkout ucx-cuda
     ./autogen.sh
     mkdir build
     cd build
     ../configure --prefix=$CONDA_PREFIX --enable-debug --with-cuda=$CUDA_HOME --enable-mt --disable-cma CPPFLAGS="-I//$CUDA_HOME/include"
```
Comment (jakirkham) [inline README.md]: I think we can drop `--disable-cma` now as that was fixed in `master`. ( https://github.com/openucx/ucx/pull/3424 )

**Comment by jakirkham** [comment]
We may want to [update the `conda install` command for non-development installs]( https://github.com/rapidsai/ucx-py/issues/241#issuecomment-541175247 ), but I'm also happy to handle this independently.

File: README.md (hunk 1 above)
Comment (VibhuJawa) [reply README.md]: Raised: https://github.com/rapidsai/dask-cuda/issues/159

Comment (jakirkham) [reply README.md]: Thanks @VibhuJawa! 😄

**Comment by jakirkham** [comment]
Have placed my other Conda related suggestions in PR ( https://github.com/rapidsai/ucx-py/pull/253 ). Please take a look and let me know what you think 🙂

File: README.md [hunk 3]
```diff
@@ -58,15 +58,11 @@ These three libraries provide a powerful combination of HPC message passing tool
 
     git clone https://github.com/openucx/ucx
     cd ucx
-    git remote add Akshay-Venkatesh git@github.com:Akshay-Venkatesh/ucx.git
-    git remote update Akshay-Venkatesh
-    git checkout ucx-cuda
```
Comment (jakirkham) [inline README.md]: Updating CI to match in PR ( https://github.com/rapidsai/ucx-py/pull/268 ).

**Pull request merged**